client/lib/layout-focus/index.js

```diff
@@ -1,9 +1,51 @@
+/** @ssr-ready **/
+
 /**
  * Internal dependencies
  */
-var Emitter = require( 'lib/mixins/emitter' ),
-	config = require( 'config' );
+import Emitter from 'lib/mixins/emitter';
+import config from 'config';
+import { getCurrentLayoutFocus, getPreviousLayoutFocus } from 'state/ui/selectors';
+import { setLayoutFocus, setNextLayoutFocus, activateNextLayoutFocus } from 'state/ui/actions';
+
+// These are the structural areas
+// of the main body of Calypso
+const _areas = [ 'content', 'sidebar', 'sites', 'preview' ];
+function isValid( area ) {
+	const valid = _areas.indexOf( area ) !== -1;
+
+	if ( ! valid ) {
+		if ( config( 'env' ) === 'development' ) {
+			throw new Error( area + ' is not a valid layout focus area' );
+		}
+	}
 
+	return valid;
+}
+
+// This is unfortunately necessary to avoid layout elements still in the DOM
+// being targeted by things like browser inline search, which may cause odd
+// positioning effects by having their CSS modified.
+function setFocusHideClass() {
+	if ( typeof document === 'undefined' ) {
+		return;
+	}
+	// Whenever layout focus changes remove `focus-hide` so
+	// that animations can occur with all elements visible.
+	document.documentElement.classList.remove( 'focus-hide' );
+
+	// After transitions restore `focus-hide`
+	setTimeout( function() {
+		document.documentElement.classList.add( 'focus-hide' );
+	}, 200 );
+}
+
+// Used to keep a reference to the Redux store so we can update it when the
+// legacy Flux store is updated.
+let reduxStore;
```

Comment:
My initial worry with binding Redux store this way is that on server side, since module references are shared, it _could_ open the possibility that a store instance is leaked between requests by different users.